test/src/specs/alert/alert_propagation.rs

```diff
@@ -25,15 +26,16 @@ impl Default for AlertPropagation {
 }
 
 impl Spec for AlertPropagation {
-    crate::name!("alert_propagation");
     crate::setup!(num_nodes: 3);
+
     // Case: alert propagation in p2p network
     //    1. create and send alert via node0; all nodes should receive the alert;
     //    2. cancel previous alert via node0; all nodes should receive the alert;
     //    3. resend the first alert, all nodes should ignore the alert.
+    fn run(&self, nodes: &mut Vec<Node>) {
```

Comment:
it's more common to pass slice as arg rather than vec since we just want to provide write access to individual elements (not whole vector)
```suggestion
    fn run(&self, nodes: &mut [Node]) {
```

Reply:
There are some code change the vector, e.g. https://github.com/nervosnetwork/ckb/blob/da59d97cd0d43954565adaab97ee8fff2e5e2218/test/src/specs/p2p/whitelist.rs#L24-L28

![image](https://user-images.githubusercontent.com/1870648/94100580-9b0f2d00-fe60-11ea-8b2e-1244f7a7d630.png)

Maybe we can optimize all the code below, but I don't like to do it right now.

Reply:
Can you confirm that a simple search and replace will work?
```diff
- let node1 = nodes.pop().unwrap();
+ let node1 = nodes[0]
```
If yes, I insist on making changes in this PR since we change interface `fn run(&mut Net)` => `fn run(&mut Vec<Node>)`, otherwise we need to change all test case again with `fn run(&mut Vec<Node>)` => `fn run(&mut [Node])` in other PR

Reply:
I try to replace `nodes.pop().unwrap()` to `&nodes[x]` / `&mut nodes[x]` just now. But all failed because of the Rust mutable requirement.
  - `&nodes[x]`: In that cases, `Node` need mutable access to involve `Node::start()` and `Node::stop()`.
  - `&mut nodes[x]`: If I borrow a mutable element, then the later borrow would error:
    ```shell
    67 |         let node0 = &mut nodes[0];
       |                          ----- mutable borrow occurs here
    68 |         let node1 = &nodes[1];
    69 |         exit_ibd_mode(nodes);
       |                       ^^^^^ immutable borrow occurs here
    ```

Reply:
i didn't realize the borrow checking here, OK, let's keep the interface to Vec